Source/GSStoryboardTransform.m

```diff
@@ -0,0 +1,835 @@
+              withDoc: (NSXMLElement *)doc
+          withAppNode: (NSXMLNode *)appNode
+              sceneId: (NSString *)sceneId
+{
+  NSString *customClassString = nil;
+  if (appNode != nil)
+    {
+      NSXMLElement *objects = (NSXMLElement *)[appNode parent];
+      NSArray *appConsArr = [appNode nodesForXPath: @"connections" error: NULL];
+      NSXMLNode *appCons = [appConsArr objectAtIndex: 0];
+
+      if (appCons != nil)
+        {
+          [appCons detach];
+        }
+      
+      // Assign application scene...
+      ASSIGN(_applicationSceneId, sceneId);
+      [_controllerMap setObject: _applicationSceneId
+                         forKey: APPLICATION];
+      
+      // Move all application children to objects...
+      NSArray *appChildren = [appNode children];      
+      FOR_IN(NSXMLElement*, ae, appChildren)
+        {
+          [ae detach];
+          [objects addChild: ae];
+        }
+      END_FOR_IN(appChildren);
+      
+      // Remove the appNode
+      [appNode detach];
+      
+      // Add it to the document
+      [objects detach];
+      [doc addChild: objects];
+
+      // create a customObject entry for NSApplication reference...
+      NSXMLNode *appCustomClass = (NSXMLNode *)[(NSXMLElement *)appNode
+                                                   attributeForName: @"customClass"];
+      customClassString = ([appCustomClass stringValue] == nil) ?
+        @"NSApplication" : [appCustomClass stringValue];
+
+      [self addStandardObjects: objects
+                   classString: customClassString
+                   connections: appCons];
+    }
+  else
+    {
+      [self addStandardObjects: child
+                   classString: customClassString
+                   connections: nil];
+      [child detach];
+      [doc addChild: child];
+    }
+}
+
+- (NSArray *) findSubclassesOf: (Class)clz
+                    inDocument: (NSXMLDocument *)document
+{
+  NSArray *result = nil;
+  NSMutableArray *subclasses = [GSObjCAllSubclassesOfClass(clz) mutableCopy];
+
+  [subclasses insertObject: clz atIndex: 0];
+  FOR_IN(Class, cls, subclasses)
+    {
+      NSString *className = NSStringFromClass(cls);
+      NSString *classNameNoNamespace = [className substringFromIndex: 2];
+      NSString *xmlClassName = [NSString stringWithFormat: @"%@%@",
+                                         [[classNameNoNamespace substringToIndex: 1] lowercaseString],
+                                         [classNameNoNamespace substringFromIndex: 1]];
+      NSString *xpath = [NSString stringWithFormat: @"//%@",xmlClassName];
+      result = [document nodesForXPath: xpath error: NULL];
+
+      if ([result count] > 0)
+        {
+          break;
+        }
+    }
+  END_FOR_IN(subclasses);
+
+  return result;
+}
+
+- (NSString *) controllerIdWithDocument: (NSXMLDocument *)document
+{
+  NSString *controllerId = nil;
+  NSArray *windowControllers = [self findSubclassesOf: [NSWindowController class]
```

Comment:
Is this search for subclasses of windowController and viewController really needed? I look through a few projects on GitHub but couldn't find any using this. I know this does not prove anything. There is a customClass attribute on the XML node, so I would not expect to see a different tag for the subclasses. Do you have any example where this gets used? 
Removing this code would again simply the process here. 
I also only found at most one controller per scene. Do you have any more information on that?

Reply:
I finally found one such subclass: navigationController, which seems to be important for UIKit. Currently we don't support that and even then it would be more useful to just add that string here directly.

Reply:
Yes, because there may be subclasses that need also to be used.  NSTabViewController and NSPageController... any subclass of NSViewController which might be added in the future.  I did this to future-proof the code.

Reply:
https://developer.apple.com/documentation/appkit/nstabviewcontroller?language=objc

https://developer.apple.com/documentation/appkit/nssplitviewcontroller?language=objc

Reply:
For instance NSSplitViewController comes up as follows in the storyboard...

`+        <scene sceneID="Qzn-Ms-zfD">
+            <objects>
+                <splitViewController id="Eua-kV-LFV" sceneMemberID="viewController">
+                    <splitViewItems>
+                        <splitViewItem id="seP-J7-ftb"/>
+                        <splitViewItem id="IZ4-Aw-WYN"/>
+                    </splitViewItems>
+                    <splitView key="splitView" dividerStyle="thin" vertical="YE:
`

Reply:
The three controllers mentioned are next on my list to be implemented, so this code will be further tested once that's done.